### ISSUE TITLE
409 conflict error

### ISSUE THREAD

**Issue by zengruizhao** [tulir/whatsmeow#382]
I always get an error when I mark mute or archive
`Error changing chat's mute state: server returned error updating app state: <collection name="regular_low" type="error"><error code="409" text="conflict"/><patches><patch><!-- 830 bytes --></patch><patch><!-- 700 bytes --></patch><patch><!-- 732 bytes --></patch><patch><!-- 750 bytes --></patch><patch><!-- 700 bytes --></patch><patch><!-- 1000 bytes --></patch><patch><!-- 660 bytes --></patch><patch><!-- 846 bytes --></patch><patch><!-- 774 bytes --></patch><patch><!-- 732 bytes --></patch><patch><!-- 1250 bytes --></patch><patch><!-- 1302 bytes --></patch><patch><!-- 894 bytes --></patch><patch><!-- 756 bytes --></patch><patch><!-- 628 bytes --></patch><patch><!-- 700 bytes --></patch><patch><!-- 732 bytes --></patch><patch><!-- 910 bytes --></patch><patch><!-- 756 bytes --></patch><patch><!-- 926 bytes --></patch><patch><!-- 854 bytes --></patch><patch><!-- 1000 bytes --></patch><patch><!-- 1222 bytes --></patch><patch><!-- 732 bytes --></patch><patch><!-- 814 bytes --></patch><patch><!-- 870 bytes --></patch></patches></collection>`

version: 2.2319.9

**Comment by zengruizhao** [comment]
I solved it through modifing the param of appstate.go from 
`mutations, newState, err := cli.appStateProc.DecodePatches(patches, state, true)`
to 
`mutations, newState, err := cli.appStateProc.DecodePatches(patches, state, false)`
That is, to cancel the execution of the method validateSnapshotMAC.
I don't know why to implement this method

**Comment by Ased2235** [comment]
Did it worked for you? for me it didn't return any error but it didn't mute it either.

**Comment by zengruizhao** [comment]
> Did it worked for you? for me it didn't return any error but it didn't mute it either.

yeah，this problem was solved by the above processing method.

**Comment by arthurrferroni** [comment]
any news on this?
facing this problem when trying to Mark as read/unread and Archive a chat

**Comment by ahmedRSA** [comment]
hey tulir any plan on fixing this 
the issue has been opened since 3 years 

this is happening so much lately and wallahi this is annoying also there is no even a way to get around it

**Comment by ahmedRSA** [comment]
> I solved it through modifing the param of appstate.go from `mutations, newState, err := cli.appStateProc.DecodePatches(patches, state, true)` to `mutations, newState, err := cli.appStateProc.DecodePatches(patches, state, false)` That is, to cancel the execution of the method validateSnapshotMAC. I don't know why to implement this method

this does not solve the problem

**Comment by tulir** [comment]
Non-fatal 409s have been auto-retried for a while now, the rest are likely #858
